### DISCUSSION TITLE
Login Persistance Flutter SDK

### DISCUSSION THREAD

**Discussion by bmartinezg** [pocketbase/pocketbase#857] [Q&A] [answered]
Hello!

Is there any way to persist the user and token in the device with the Flutter SDK? Currently I am using the authStore service but every time i refresh the app, the param isValid inside authStore is false. 

I would like to be able to evaluate that params (isValid) and if is true, redirect to the home screen and if is false, to the login screen.

Thank

**Comment by ganigeorgiev** [comment, marked as answer]
The Dart SDK currently doesn't have persistence storage since it is intended to be a generic Dart package not just for Flutter and besides, as mentioned in https://github.com/pocketbase/dart-sdk/issues/13, for Flutter there are too many available options (Localstore, Sharedpreferences/EncryptedSharedPreferences, various key-value file dbs, etc.) and it is maybe better to leave the developers decide what is best for their app.

You can create your own `AuthStore` by extending the default one and set it during client initialization as shown in https://github.com/pocketbase/dart-sdk/#authstore. Basically, you have to provide `save` and `clear` implementations extending https://github.com/pocketbase/dart-sdk/blob/master/lib/src/auth_store.dart.

**Reply by rodydavis** [reply]
Planning on adding it to my **pocketbase_drift** package (backed by local SQLite). 
https://github.com/rodydavis/pocketbase_drift/pull/8

**Reply by ganigeorgiev** [reply]
_Just for info, you can also find an example that you can use as a reference for a custom store in https://github.com/pocketbase/pocketbase/discussions/1887#discussioncomment-5057297 (_for simplicity it uses shared_preferences, but it can be anything else; additionally it may need an internal queue to serialize the read/write operations since the base `AuthStore` is not async_). The refactoring of both JS and Dart SDKs will be prioritized after the upcoming v0.17.0 release and I'll consider adding proper React Native and Flutter AuthStore examples._

**Reply by rodydavis** [reply]
Added it and works well! I am saving to a SQLite table instead
